src/mngt/ocf_mngt_cache.c

```diff
@@ -1290,6 +1273,43 @@ static void _ocf_mngt_init_attached_nonpersistent(ocf_cache_t cache)
 	env_atomic_set(&cache->fallback_pt_error_counter, 0);
 }
 
+static void _ocf_mngt_copy_uuid_data(ocf_pipeline_t pipeline,
+		void *priv, ocf_pipeline_arg_t arg)
+{
+	struct ocf_cache_attach_context *context = priv;
+	struct ocf_mngt_cache_device_config *cfg = &context->cfg;
+	_ocf_mngt_cache_attach_end_t cmpl = context->cmpl;
+	ocf_cache_t cache = context->cache;
+	void *priv1 = context->priv1;
+	void *priv2 = context->priv2;
+	void *data;
+	int result;
+
+	data = env_vmalloc(cfg->uuid.size);
+	if (!data) {
+		result = -OCF_ERR_NO_MEM;
+		goto err_pipeline;
+	}
+
+	result = env_memcpy(data, cfg->uuid.size, cfg->uuid.data,
+			cfg->uuid.size);
+	if (result)
+		goto err_uuid;
+
+	context->cfg.uuid.data = data;
+
+	_ocf_mngt_init_attached_nonpersistent(cache);
+
+	OCF_PL_NEXT_RET(pipeline);
+
+err_uuid:
+	env_vfree(data);
+err_pipeline:
+	ocf_pipeline_destroy(pipeline);
+	ocf_pipeline_destroy(cache->stop_pipeline);
+	OCF_CMPL_RET(cache, priv1, priv2, result);
+}
+
```

Comment:
you should call OCF_PL_FINISH_RET(pipeline, result) instead of deallocating pipeline manually - this made sense when the code was executed outside of pipeline context

Reply:
Also be careful with freeing `data` in pipeline finish, because if error occurred in line 1299 it will result in double free. Setting `context->cfg.uuid.data` in error path will be probably the simples solution.